asset-transfer-basic/application-javascript/app.js

```diff
@@ -7,127 +7,149 @@
+			const network = await gateway.getNetwork(channelName);
+
+			// Get the contract from the network.
+			const contract = network.getContract(chaincodeName);
+
+			// Initialize a set of asset data on the channel using the chaincode 'InitLedger' function.
+			// This type of transaction would only be run once by an application the first time it was started after it
+			// deployed the first time. Any updates to the chaincode deployed later would likely not need to run
+			// an "init" type function.
+			console.log('\n--> Submit Transaction: InitLedger, function creates the initial set of assets on the ledger');
+			await contract.submitTransaction('InitLedger');
+			console.log('*** Result: committed');
+
+			// Let's try a query type operation (function).
+			// This will be sent to just one peer and the results will be shown.
+			console.log('\n--> Evaluate Transaction: GetAllAssets, function returns all the current assets on the ledger');
+			let result = await contract.evaluateTransaction('GetAllAssets');
+			console.log(`*** Result: ${prettyJSONString(result.toString())}`);
+
+			// Now let's try to submit a transaction.
+			// This will be sent to both peers and if both peers endorse the transaction, the endorsed proposal will be sent
+			// to the orderer to be committed by each of the peer's to the channel ledger.
+			console.log('\n--> Submit Transaction: CreateAsset, creates new asset with ID, color, owner, size, and appraisedValue arguments');
+			await contract.submitTransaction('CreateAsset', 'asset13', 'yellow', '5', 'Tom', '1300');
+			console.log('*** Result: committed');
+
+			console.log('\n--> Evaluate Transaction: ReadAsset, function returns an asset with a given assetID');
+			result = await contract.evaluateTransaction('ReadAsset', 'asset13');
+			console.log(`*** Result: ${prettyJSONString(result.toString())}`);
+
+			console.log('\n--> Evaluate Transaction: AssetExists, function returns "true" if an asset with given assetID exist');
+			result = await contract.evaluateTransaction('AssetExists', 'asset1');
+			console.log(`*** Result: ${prettyJSONString(result.toString())}`);
+
+			console.log('\n--> Submit Transaction: UpdateAsset asset1, change the appraisedValue to 350');
+			await contract.submitTransaction('UpdateAsset', 'asset1', 'blue', '5', 'Tomoko', '350');
+			console.log('*** Result: committed');
+
+			console.log('\n--> Evaluate Transaction: ReadAsset, function returns "asset1" attributes');
+			result = await contract.evaluateTransaction('ReadAsset', 'asset1');
+			console.log(`*** Result: ${prettyJSONString(result.toString())}`);
+
+			try {
+				// How about we try a transactions where the executing chaincode throws an error
+				// Notice how the submitTransaction will throw an error containing the error thrown by the chaincode
+				console.log('\n--> Submit Transaction: UpdateAsset asset70, asset70 does not exist and should return an error');
```

Comment:
Thank you! I was going to suggest to make that change to indicate the error is expected. :-)